pkg/kubecost/allocation.go

```diff
@@ -267,26 +291,77 @@ func (parc ProportionalAssetResourceCost) Key(insertByNode bool) string {
 
 type ProportionalAssetResourceCosts map[string]ProportionalAssetResourceCost
 
+func (parcs ProportionalAssetResourceCosts) Clone() ProportionalAssetResourceCosts {
+	cloned := ProportionalAssetResourceCosts{}
+
+	for key, parc := range parcs {
+		cloned[key] = parc.Clone()
```

Comment:
For values, I think you shouldn't need to clone here, and that may result in better performance. Cloning is required when you have pointers, not values, and want to avoid wacky state mutation bugs.

Reply:
I think Go does some fancy compiler stuff when it sees a for loop and values populating a map. But the Clone call probably breaks that.

Reply:
thanks, fixed!